compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java

```diff
@@ -846,8 +848,8 @@ private void generateRefNotEqualIns(BIRNonTerminator.BinaryOp binaryIns) {
         } else if (lhsOpType.tag == TypeTags.BYTE && rhsOpType.tag == TypeTags.BYTE) {
             this.mv.visitJumpInsn(IF_ICMPEQ, label1);
         } else if (lhsOpType.tag == TypeTags.FLOAT && rhsOpType.tag == TypeTags.FLOAT) {
-            this.mv.visitInsn(DCMPL);
-            this.mv.visitJumpInsn(IFEQ, label1);
+            this.mv.visitMethodInsn(INVOKESTATIC, TYPE_CHECKER, CHECK_FLOAT_REF_EQUAL, "(DD)Z", false);
```

Comment:
This is ok. but it is faster to actually generate the check here.

Reply:
Current way is ok for this PR.

Reply:
Shall we directly generate check here.

Reply:
Lets add that improvement with Fix for https://github.com/ballerina-platform/ballerina-lang/issues/17977